src/crimson/common/logfile.cc

```diff
@@ -0,0 +1,88 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2011 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include <fstream>
+#include <seastar/util/log.hh>
+
+#include "common/debug.h"
+#include "logfile.h"
+#include "log.h"
+
+#define dout_context g_ceph_context
+#define dout_prefix *_dout
+#define dout_subsys ceph_subsys_
+
+namespace {
+  seastar::logger& logger() {
+    return crimson::get_logger(ceph_subsys_osd);
+  }
+}  // namespace
+
+struct logfh {
+  std::string log_file_path;
+  std::ofstream log_ofstream;
+
+//  logfh() {
+//    log_file_path.clear();
+//  }
+
+  ~logfh() {
+    log_ofstream.flush();
+    log_ofstream.close();
+  }
+
+  bool is_open() const {
+    return !log_file_path.empty() && log_ofstream.is_open();
+  }
+  int open(std::string_view log_file);
+};
+
+std::unique_ptr<logfh> lfh;
+int logfh::open(std::string_view log_file)
+{
+  log_file_path = log_file;
+
+  log_ofstream.open(log_file_path, std::ios::app | std::ios::out);
+  if (log_ofstream.fail()) {
+     return -errno;
+  }
+
+  return 0;
+}
+
+void logfile_close()
+{
+  lfh = nullptr;
+}
+
+int logfile_set_ostream(std::string_view log_file)
+{
+  if (log_file.empty()) {
+    return 0;
```

Comment:
Which means a success? is this the expected ret val? what would happen next?

Reply:
in case of empty log_file or missing config of log_file we are logging to the stdout\stderr as we currently doing.